ModuleFromCriterion.lua

```diff
@@ -24,7 +24,19 @@ end
 function ModuleFromCriterion:updateGradInput(input, gradOutput)
    local prediction, target = unpack(input)
    local gradPrediction = self.criterion:updateGradInput(prediction, target)
-   self.gradInput[1]:resizeAs(gradPrediction):copy(gradPrediction):mul(gradOutput[1])
+   if type(gradPrediction) == 'table' then
+      if type(self.gradInput[1]) ~= 'table' then
+         self.gradInput[1] = {} -- initializing to table first time if it is tensor (which it is: line 10)
+         for i=1, #gradPrediction do
+            self.gradInput[1][i] = gradPrediction[i]:clone() -- and putting tensors of right size inside.
```

Comment:
the clone is wasteful here. you just need:
self.gradInput[1][i] = gradPrediction[i].new()

below, you resize the tensor and copy the contents over.